pkg/repl/subscription_map.go

```diff
@@ -89,21 +89,31 @@ func (s *deltaMap) createReplaceStmt(replaceKeys []string) statement {
 	}
 }
 
-// Flush writes changes to the new table.
-// If underLock is true, then it uses the provided lock to execute
-// the statements under a table lock. This is used for the final flush
-// to ensure no changes are missed.
-func (s *deltaMap) Flush(ctx context.Context, underLock bool, lock *dbconn.TableLock) error {
+// Flush writes the pending changes to the new table.
+// We do this under a mutex, which means that unfortunately pending changes
+// are blocked from being collected while we do this. In future we may
+// come up with a more sophisticated approach to allow concurrent
+// collection of changes while we flush.
+func (s *deltaMap) Flush(ctx context.Context, underLock bool, lock *dbconn.TableLock) (allChangesFlushed bool, err error) {
 	s.Lock()
 	defer s.Unlock()
 
 	var deleteKeys []string
 	var replaceKeys []string
 	var stmts []statement
+	var keysFlushed []string // keys we need to remove from the map at the end.
 	var i int64
+	allChangesFlushed = true // assume all changes are flushed unless we find some that are not.
 	target := atomic.LoadInt64(&s.c.targetBatchSize)
 	for key, isDelete := range s.changes {
+		unhashedKey := utils.UnhashKey(key)
+		if s.chunker != nil && !s.chunker.KeyBelowLowWatermark(unhashedKey[0]) {
+			s.c.logger.Debugf("key below watermark: %v", unhashedKey[0])
```

Comment:
This appears to check that the key is *not* below the watermark (`!s.chunker.KeyBelowLowWatermark`), but the log line says the opposite?